lib/membrane_live/event.ex

```diff
@@ -0,0 +1,239 @@
+       event_id: event_id,
+       rtc_engine: pid,
+       peer_channels: %{},
+       network_options: network_options,
+       trace_ctx: trace_ctx
+     }}
+  end
+
+  @impl true
+  def handle_info({:add_peer_channel, peer_channel_pid, peer_id}, state) do
+    state = put_in(state, [:peer_channels, peer_id], peer_channel_pid)
+    Process.monitor(peer_channel_pid)
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(%Message.MediaEvent{to: :broadcast, data: data}, state) do
+    for {_peer_id, pid} <- state.peer_channels, do: send(pid, {:media_event, data})
+
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(%Message.MediaEvent{to: to, data: data}, state) do
+    if state.peer_channels[to] != nil do
+      send(state.peer_channels[to], {:media_event, data})
+    end
+
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(%Message.NewPeer{rtc_engine: rtc_engine, peer: peer}, state) do
+    Membrane.Logger.info("New peer: #{inspect(peer)}. Accepting.")
+    peer_channel_pid = Map.get(state.peer_channels, peer.id)
+    peer_node = node(peer_channel_pid)
+
+    handshake_opts =
+      if state.network_options[:dtls_pkey] &&
+           state.network_options[:dtls_cert] do
+        [
+          client_mode: false,
+          dtls_srtp: true,
+          pkey: state.network_options[:dtls_pkey],
+          cert: state.network_options[:dtls_cert]
+        ]
+      else
+        [
+          client_mode: false,
+          dtls_srtp: true
+        ]
+      end
+
+    endpoint = %WebRTC{
+      rtc_engine: rtc_engine,
+      ice_name: peer.id,
+      owner: self(),
+      integrated_turn_options: state.network_options[:integrated_turn_options],
+      integrated_turn_domain: state.network_options[:integrated_turn_domain],
+      handshake_opts: handshake_opts,
+      log_metadata: [peer_id: peer.id],
+      trace_context: state.trace_ctx,
+      webrtc_extensions: [Mid, TWCC],
+      peer_metadata: peer.metadata,
+      filter_codecs: fn {rtp, fmtp} ->
+        case rtp.encoding do
+          "opus" -> true
+          "H264" -> fmtp.profile_level_id === 0x42E01F
+          _unsupported_codec -> false
+        end
+      end
+    }
+
+    Engine.accept_peer(rtc_engine, peer.id)
+    :ok = Engine.add_endpoint(rtc_engine, endpoint, peer_id: peer.id, node: peer_node)
+
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(%Message.PeerLeft{peer: peer}, state) do
+    Membrane.Logger.info("Peer #{inspect(peer.id)} left RTC Engine")
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(%Message.EndpointCrashed{endpoint_id: endpoint_id}, state) do
+    Membrane.Logger.error("Endpoint #{inspect(endpoint_id)} has crashed!")
+    peer_channel = state.peer_channels[endpoint_id]
+
+    error_message = "WebRTC endpoint has crashed, please refresh the page to reconnect"
+    data = MediaEvent.create_error_event(error_message)
+    send(peer_channel, {:media_event, data})
+
+    {:noreply, state}
+  end
+
```

Comment:
As we could have different types you could change the error message and add some TODO as we probably will have handle different way for crash WebRTC than crash of HLS Endpoint.